core/platform/search/elastic_search_services.py

```diff
@@ -145,9 +142,35 @@ def search(
     """
     assert cursor is None
     assert offset + size < MAXIMUM_NUMBER_OF_RESULTS
-    query_definiton = json.loads(query_string)
+
```

Comment:
is there any issue with fetching categories and language_codes from the query string over here? (I think if we are having query_string param then `categories and language_codes` should be inside the query string as an API OR replace query_string with something specific it can have.)